test/functional/cmdline_options_tester/src/Tokenizer.java

```diff
@@ -81,30 +81,36 @@ private String token() throws Exception {
          * otherwise it will ends up returning an empty argument to be ignored in forming the command array.
          *
          * Three cases are taken into account in the while loop:
-         * 1) Strings in double-quotes (a argument starting with double-quotes(")) are completely extracted from the command string,
-         *    whether white spaces in the argument exist or not. Double-quotes themselves are discarded after extraction when scanning forward.
-         * 2) Normal arguments (no double-quotes occur in the argument) are split up and extracted with white spaces between arguments.
+         * 1) Strings in double-quotes/single-quotes (a argument starting with double-quotes(") or single-quotes(')) are completely extracted from the command string,
+         *    whether white spaces in the argument exist or not. Double-quotes/single-quotes themselves are discarded after extraction when scanning forward.
+         * 2) Normal arguments (no double-quotes/single-quotes occur in the argument) are split up and extracted with white spaces between arguments.
          * 3) Following the original order of the command line, non-quoted portions of an argument would be extracted in case 2)
          *    and concatenated to quoted portions of the argument extracted in case 1) within the external while loop.
          */
         while ((index < buffer.length()) && !Character.isWhitespace(buffer.charAt(index))) {
 
         	/**
-        	 * Case 1: Expect a double-quoted string once a double-quotes is detected
+        	 * Case 1: Expect a double-quoted or single-quoted string once either is detected
         	 */
-            if (buffer.charAt(index) == '"') {
+            if (buffer.charAt(index) == '"' || buffer.charAt(index) == '\'') {
                 index++;
                 int i = index;
+                char delim;
+                if (buffer.charAt(index-1) == '"') {
+                	delim = '"';
+                } else{
+                	delim = '\'';
+                }
```

Comment:
Perhaps `delim` should be moved up before the `if`:
```
char delim = buffer.charAt(index);
if ((delim == '"') || (delim == '\'')) {
```
Note the extra parentheses (so a reader doesn't have to consider the relative precedence of `==` vs. `||`).

Reply:
Also fixed!  I should have done it that way to begin with.  Thanks for pointing this out!